### ISSUE TITLE
SystemRealTimeMsg::SystemReset cannot be parsed from to_midi()

### ISSUE THREAD

**Issue by Dicklessgreat** [AlexCharlton/midi-msg#24]
```
let system_reset = MidiMsg::SystemRealTime {
    msg: SystemRealTimeMsg::SystemReset,
};
assert_eq!(
    MidiMsg::from_midi_with_context(&system_reset.to_midi(), &mut ReceiverContext::new()),
    Ok((system_reset, 1)),
);
```
run result:
```
assertion `left == right` failed
  left: Err(UnexpectedEnd)
 right: Ok((SystemRealTime { msg: SystemReset }, 1))
```
It seems to be[ interpreting it as a `Meta` message](https://github.com/AlexCharlton/midi-msg/blob/f1e3126abb12e5b83a99fcbedb2b6f61129f713c/src/message.rs#L169) and[ returning `Err(ParseError::UnexpectedEnd)`](https://github.com/AlexCharlton/midi-msg/blob/f1e3126abb12e5b83a99fcbedb2b6f61129f713c/src/file.rs#L626) because it has no more than 2 bytes of data.

Can we do better?:)

**Comment by AlexCharlton** [comment]
Yeah, this is a nasty bug! Thanks for the report. Will take a look...

**Comment by AlexCharlton** [comment]
Fixed with https://github.com/AlexCharlton/midi-msg/commit/7d840bb79d2d63fbbdd68283fb6748f2266c6ae6 and published a new version 0.7.2.

The key to the fix is that `SystemReset` messages cannot exist in SMF, since they overlap with Meta messages. And now that I write this, I'm realizing that I should throw an error if someone tries to add a Reset to an SMF... I'll do that too 😅

**Comment by Dicklessgreat** [comment]
updated my crate to `0.7.2` and confirmed it works fine!
Thanks!
